src/wasm/wasm-binary.cpp

```diff
@@ -2464,7 +2464,12 @@ void WasmBinaryReader::readTypes() {
   if (auto* err = result.getError()) {
     Fatal() << "Invalid type: " << err->reason << " at index " << err->index;
   }
-  types = *result;
+  types = std::move(*result);
+
+  // Record the type indices.
+  for (Index i = 0; i < types.size(); ++i) {
+    wasm.typeIndices.insert({types[i], i});
+  }
```

Comment:
Is there a performance downside to this extra work?

Reply:
I would be shocked if it were measurable, but I can check.

Reply:
It's worth measuring I think, to be sure. I was imagining this would be behind the flag, but if there is no speed benefit and it is cumbersome to do so, then it's fine as it is.

Reply:
I measured, and indeed this was in the noise.